### ISSUE TITLE
Hello! Could you tell me how to save and load model's weights in ".pt" file?

### ISSUE THREAD

**Issue by 943fansi** [yonda-yonda/tadgan#1]
@yonda-yonda Thanks very much for your code. I am using this for series anomaly detection on my own data. I want to save the model's weights and load it. Could you help with me?

**Comment by yonda-yonda** [comment]
@943fansi Thanks for your interest.

Please try to use [ProcessedDataset](https://github.com/yonda-yonda/tadgan/blob/b5c0a8323cbe1ef91c989f53f30dbaaddb5f12ad/tadgan/tadgan.py#L99).

For example,  see [processed_train_A1.csv](https://github.com/yonda-yonda/tadgan/files/10180508/processed_train_A1.csv). This is telemanom's data processed by [me](https://github.com/yonda-yonda/tadgan/blob/b5c0a8323cbe1ef91c989f53f30dbaaddb5f12ad/tadgan/telemanom.py#L7).

This data are [normalized](https://github.com/yonda-yonda/tadgan/blob/b5c0a8323cbe1ef91c989f53f30dbaaddb5f12ad/tadgan/telemanom.py#L13), and `window_size=100`, `step_size=1`.

If you create a csv like this, all you have to do is read it by DatasetClass extends ProcessedDataset.

**Comment by 943fansi** [comment]
@yonda-yonda Thank you for your answer. I apologize for not expressing my question clearly. I have completed the preprocessing and am able to train the model successfully. My requirement is to train and save the model on one computer, and then perform inference on another computer. In other words, the training and prediction are separated. For a typical network, I usually do the following:
`net = MLP()`
###### after training
`torch.save(net.state_dict(), "net.pth")`
`state_dict = torch.load("net.pth")`
`model = MLP()`
 `model.load_state_dict(state_dict)`
How do I save the trained TadGAN model and load the network weights?

**Comment by yonda-yonda** [comment]
@943fansi I understand.
And sorry, I didn't care about save and load model.  Surely that's necessary.
Give me a few days to think.

**Comment by yonda-yonda** [comment]
https://github.com/yonda-yonda/tadgan/pull/2
Is it what you want?

**Comment by 943fansi** [comment]
@yonda-yonda Yeah. Thanks very much. It solved my problem very well. Do you have any good suggestions and methods for threshold setting of abnormal score?

**Comment by yonda-yonda** [comment]
@943fansi I would have to merge it.
My tadgan is so much better, thanks to your comment.

About threshold settings, see simple comparisons with different parameters (sorry for Japanese text).
https://zenn.dev/yonda/articles/681f32264341cf#a1

So `anomaly_score` is zscore, if you want to evaluate conservatively, the threshold should be 3.
However, we have to solve each problem accordingly in the end.

The [implementation](https://github.com/sintel-dev/Orion/blob/6548161e240a413ba011a699d109eb8a1e50c148/orion/primitives/tadgan.py) by the author of the paper has some innovations that are not in my code, so I recommend that you refer to that as well.

**Comment by 943fansi** [comment]
@yonda-yonda Thanks!
